app/controllers/spree/admin/products_controller.rb

```diff
@@ -1,30 +1,86 @@
+require 'open_food_network/spree_api_key_loader'
+require 'open_food_network/referer_parser'
+require 'open_food_network/permissions'
+
 module Spree
   module Admin
     class ProductsController < ResourceController
       helper 'spree/products'
+      include OpenFoodNetwork::SpreeApiKeyLoader
+      include OrderCyclesHelper
+      include EnterprisesHelper
 
       before_filter :load_data, :except => :index
       create.before :create_before
       update.before :update_before
 
+      before_filter :load_data
+      before_filter :load_form_data, only: [:index, :new, :create, :edit, :update]
+      before_filter :load_spree_api_key, only: [:index, :variant_overrides]
+      before_filter :strip_new_properties, only: [:create, :update]
+
+      respond_override create: { html: {
+        success: lambda {
+          if params[:button] == "add_another"
+            redirect_to new_admin_product_path
+          else
+            redirect_to admin_products_path
+          end
+        },
+        failure: lambda {
+          render :new
+        }
+      } }      
+
+      def new
+        @object.shipping_category = DefaultShippingCategory.find_or_create
+        super
+      end
+
+      def create
+        delete_stock_params_and_set_after do
+          super
```

Comment:
Just to clarify. This `super` was in the override before. And it called `RessourceController.create`, not the overridden `ProductsController.create`, right? In that case, this is all good. :-) This confusion is a good reason to get rid of these overrides.

Reply:
yes, super is not the class overridden by class_eval but rather the class overridden by inheritance.